pkg/commands/helpers/kubectl_auth.go

```diff
@@ -174,20 +173,8 @@ func cacheFilePath(kopsStateStore string, clusterName string) string {
 	b.WriteString(clusterName)
 	b.WriteByte(0)
 
-	hash := fmt.Sprintf("%x", sha256.New().Sum(b.Bytes()))
-	sanitizedName := strings.Map(func(r rune) rune {
-		switch {
-		case r >= 'a' && r <= 'z':
-			return r
-		case r >= 'A' && r <= 'Z':
-			return r
-		case r >= '0' && r <= '9':
-			return r
-		default:
-			return '_'
-		}
-	}, clusterName)
-	return filepath.Join(homedir.HomeDir(), ".kube", "cache", "kops-authentication", sanitizedName+"_"+hash)
+	hash := fmt.Sprintf("%x", sha256.Sum256(b.Bytes()))
```

Comment:
This should reduce the size of the hash a little. Should allow to keep at least a partial sanitized name.
WDYT @justinsb?
```suggestion
	h := md5.New()
	h.Write([]byte(kopsStateStore))
	h.Write([]byte(clusterName))
	hash := hex.EncodeToString(h.Sum(nil))
```

Reply:
Yes, the worry is that now the scanners will start false-positive triggering on "uses md5".  I have yet to see a tool smart enough to detect a truncation of a hash though :-)